qa/rpc-tests/sprout_sapling_migration.py

```diff
@@ -14,25 +14,40 @@
 SAPLING_ADDR = 'zregtestsapling1ssqj3f3majnl270985gqcdqedd9t4nlttjqskccwevj2v20sc25deqspv3masufnwcdy67cydyy'
 SAPLING_KEY = 'secret-extended-key-regtest1qv62zt2fqyqqpqrh2qzc08h7gncf4447jh9kvnnnhjg959fkwt7mhw9j8e9at7attx8z6u3953u86vcnsujdc2ckdlcmztjt44x3uxpah5mxtncxd0mqcnz9eq8rghh5m4j44ep5d9702sdvvwawqassulktfegrcp4twxgqdxx4eww3lau0mywuaeztpla2cmvagr5nj98elt45zh6fjznadl6wz52n2uyhdwcm2wlsu8fnxstrk6s4t55t8dy6jkgx5g0cwpchh5qffp8x5'
 
+DISABLED_NO_FUNDS = 0
+ENABLED_NO_FUNDS = 1
+DISABLED_BEFORE_MIGRATION = 2
+ENABLED_BEFORE_MIGRATION = 3
+DURING_MIGRATION = 4
+AFTER_MIGRATION = 5
+ALL_MIGRATION_STATES = [DISABLED_NO_FUNDS, ENABLED_NO_FUNDS, DISABLED_BEFORE_MIGRATION, ENABLED_BEFORE_MIGRATION, DURING_MIGRATION, AFTER_MIGRATION]
 
-def check_migration_status(
-        node,
-        enabled,
-        destination_address,
-        non_zero_unmigrated_amount,
-        non_zero_unfinalized_migrated_amount,
-        non_zero_finalized_migrated_amount,
-        finalized_migration_transactions,
-        len_migration_txids
-):
+
+def check_migration_status(node, destination_address, migration_state):
     status = node.z_getmigrationstatus()
-    assert_equal(enabled, status['enabled'])
-    assert_equal(destination_address, status['destination_address'])
-    assert_equal(non_zero_unmigrated_amount, Decimal(status['unmigrated_amount']) > Decimal('0.00'))
-    assert_equal(non_zero_unfinalized_migrated_amount, Decimal(status['unfinalized_migrated_amount']) > Decimal('0'))
-    assert_equal(non_zero_finalized_migrated_amount, Decimal(status['finalized_migrated_amount']) > Decimal('0'))
-    assert_equal(finalized_migration_transactions, status['finalized_migration_transactions'])
-    assert_equal(len_migration_txids, len(status['migration_txids']))
+    assert_equal(destination_address, status['destination_address'], "Migration destination address")
+    assert_true(migration_state in ALL_MIGRATION_STATES, "Unexpected migration state")
+
+    expected_enabled = migration_state not in [DISABLED_NO_FUNDS, DISABLED_BEFORE_MIGRATION]
+    expected_sprout_funds = migration_state in [DISABLED_BEFORE_MIGRATION, ENABLED_BEFORE_MIGRATION]
+    positive_unfinalized_amount = migration_state == DURING_MIGRATION
+    positive_finalized_amount = migration_state == AFTER_MIGRATION
+    num_migration_txids = 1 if migration_state in [DURING_MIGRATION, AFTER_MIGRATION] else 0
```

Comment:
This doesn't test that we're able to send more than one migration transaction (if notes of the right values are available).

That's a preexisting issue aside from the nondeterminism in the test, though, so please just file another ticket to improve the test coverage.